documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java

```diff
@@ -419,9 +420,17 @@ private List<FullTreeItem> getAllFolderInNode(Node node) throws RepositoryExcept
                                              folderChildListNodes));
       }
     }
-    return folderListNodes;
+    return folderListNodes.stream().sorted( new Comparator<FullTreeItem>() {
+      public int compare(FullTreeItem o1, FullTreeItem o2) {
+        //sorted the tree item when the name contains numbers
+        if(NumberUtils.isNumber(o1.getName())&&NumberUtils.isNumber(o2.getName())){
+          return Integer.parseInt(o1.getName()) - Integer.parseInt(o2.getName());
+        }
+        // sorted the tree item when the name contains only characters
+        return o1.getName().compareToIgnoreCase(o2.getName());
+      }
```

Comment:
This will sort folders by Characters AND will sort the folders which names are just numbers accordingly (1 - 2 - 3 - .... - 10 - 11 - 12 .....)
I didn't see that this was the expected behavior, but I do like it.
@hghorbel WDYT ?

Reply:
Ok for me too.

Reply:
![Screenshot from 2022-08-09 13-53-05](https://user-images.githubusercontent.com/27963909/183652219-d6e81a6a-60ed-40e8-a3da-380639a4facb.png)
@ahamdi this will sorted also folders when the name contains characters and number i think it sorted by the compareToIgnoreCase . see attachments